src/libraries/Common/src/Interop/Android/System.Security.Cryptography.Native.Android/Interop.Bignum.cs

```diff
@@ -0,0 +1,94 @@
+using System.Security.Cryptography;
+using Microsoft.Win32.SafeHandles;
+
+internal static partial class Interop
+{
+    // TODO: [AndroidCrypto] Rename class to AndroidCrypto once all consumers are split in Android vs. Unix
+    internal static partial class Crypto
+    {
+        [DllImport(Libraries.CryptoNative, EntryPoint = "AndroidCryptoNative_BigNumFromBinary")]
+        private static extern unsafe SafeBignumHandle BigNumFromBinary(byte* s, int len);
+
+        [DllImport(Libraries.CryptoNative, EntryPoint = "AndroidCryptoNative_BigNumToBinary")]
+        private static extern unsafe int BigNumToBinary(SafeBignumHandle a, byte* to);
+
+        [DllImport(Libraries.CryptoNative, EntryPoint = "AndroidCryptoNative_GetBigNumBytes")]
+        private static extern int GetBigNumBytes(SafeBignumHandle a);
+
+        internal static SafeBignumHandle CreateBignum(ReadOnlySpan<byte> bigEndianValue)
+        {
+            unsafe
+            {
+                fixed (byte* pBigEndianValue = bigEndianValue)
+                {
+                    SafeBignumHandle ret = BigNumFromBinary(pBigEndianValue, bigEndianValue.Length);
+                    if (ret.IsInvalid)
+                    {
+                        ret.Dispose();
+                        throw new CryptographicException();
+                    }
+
+                    return ret;
+                }
+            }
+        }
+
+        internal static byte[]? ExtractBignum(IntPtr bignum, int targetSize)
+        {
+            // Given that the only reference held to bignum is an IntPtr, create an unowned SafeHandle
+            // to ensure that we don't destroy the key after extraction.
+            using (SafeBignumHandle handle = new SafeBignumHandle(bignum, ownsHandle: false))
+            {
+                return ExtractBignum(handle, targetSize);
+            }
+        }
+
+        internal static unsafe byte[]? ExtractBignum(SafeBignumHandle? bignum, int targetSize)
+        {
+            if (bignum == null || bignum.IsInvalid)
+                return null;
+
+            int compactSize = GetBigNumBytes(bignum);
+            if (targetSize < compactSize)
+            {
+                targetSize = compactSize;
+            }
+
+            // Android does not include leading zeroes (uses minimum number of bytes required to represent the value).
+            // Windows Crypt32 does.
+            //
+            // Since RSACryptoServiceProvider already checks that RSAParameters.DP.Length is
+            // exactly half of RSAParameters.Modulus.Length, we need to left-pad (big-endian)
+            // the array with zeroes.
+            int offset = targetSize - compactSize;
+            byte[] buf = new byte[targetSize];
+            fixed (byte* to = buf)
+            {
+                byte* start = to + offset;
+                BigNumToBinary(bignum, start);
+            }
+
+            return buf;
+        }
+    }
+}
+
+namespace Microsoft.Win32.SafeHandles
```

Comment:
Putting the SafeHandles in Microsoft.Win32.SafeHandles is a bit of a legacy decision.  I'd go with System.Security.Cryptography, or just a nested type in Interop.[Android]Crypto.